### PR TITLE
feat: convert to TypeScript

### PR DESCRIPTION
**Pull request by alexander-fenster**
This one is complicated. It has a very weird dark magic with a helper replacing one of the auto-generated method:

https://github.com/googleapis/nodejs-speech/blob/cace910113a00217369fa1a07a5e5a96b45371e7/src/index.js#L53-L57

I'm replacing it with an equal (but probably not so dark) magic in TypeScript, making kind of a [mixin](http://www.typescriptlang.org/docs/handbook/mixins.html).

https://github.com/googleapis/nodejs-speech/blob/b4c87e7d771070aab828462de0e2eb407c1c6ad9/src/index.ts#L19-L36

https://github.com/googleapis/nodejs-speech/blob/b4c87e7d771070aab828462de0e2eb407c1c6ad9/src/v1/speech_client.ts#L477-L478

Other than that, looks good to me, and all the system tests just work.

### CODE REVIEW

**Comment by alexander-fenster** [comment]
(figuring out how to make samples linted correctly)

File: package.json [hunk 1]
```diff
@@ -29,40 +29,49 @@
...
+    "cover": "nyc --reporter=lcov mocha build/test/*.js && nyc report",
     "docs": "jsdoc -c .jsdoc.js",
-    "lint": "eslint '**/*.js'",
+    "lint": "gts check && eslint '**/*.js'",
     "samples-test": "cd samples/ && npm link ../ && npm test && cd ../",
-    "system-test": "mocha system-test/*.js --timeout 600000",
-    "test-no-cover": "mocha test/*.js",
+    "system-test": "mocha build/system-test/*.js --timeout 600000",
+    "test-no-cover": "mocha build/test/*.js",
     "test": "npm run cover",
```
Comment (bcoe) [inline package.json]: can we just replace this with `c8 mocha build/test/*.js`, it seems like this has weird bespoke coverage logic (perhaps this was the first module we added coverage to).

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import {ImprovedStreamingClient} from './helpers';
+import * as v1p1beta1 from './v1p1beta1';
+import * as v1 from './v1';
+
+// The following code is adapted from http://www.typescriptlang.org/docs/handbook/mixins.html
+// tslint:disable-next-line no-any
+Object.defineProperty(
+  v1.SpeechClient.prototype,
+  'streamingRecognize',
```
Comment (bcoe) [inline src/index.ts]: What are the main benefits of the `ImprovedStreamingClient` mix-in? better types in the response?

I wonder if we should consider upstreaming some of this to the generator itself, worth making a tracking ticket?

Comment (alexander-fenster) [reply src/index.ts]: This looks like a speech-specific change. They want `streamingRecognize`, which is a bi-di streaming method, to accept a separate config object:

```
  streamingRecognize(
    streamingConfig?:
      | protosTypes.google.cloud.speech.v1.IStreamingRecognitionConfig
      | protosTypes.google.cloud.speech.v1p1beta1.IStreamingRecognitionConfig,
    options?: gax.CallOptions
  )
```

By our design, bi-directional streaming calls don't accept any request as a parameter (only as a stream). This one is unique.

**Pull request merged**